app/Http/Controllers/UserController.php

```diff
@@ -1,10 +1,65 @@
 <?php
 
+declare(strict_types=1);
+
 namespace App\Http\Controllers;
 
-use Illuminate\Http\Request;
+use App\Http\Requests\StoreUser;
+use App\Http\Requests\UpdateUser;
+use App\Services\RegistrationService;
+use App\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
 
 class UserController extends Controller
 {
-    //
+    /**
+     * Lists all resources.
+     */
+    public function index(): JsonResponse
+    {
+        $users = User::all();
+
+        return response()->json($users, Response::HTTP_OK);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(StoreUser $request, RegistrationService $service): JsonResponse
+    {
+        $response = $service->registerUser($request->validated());
+
+        if (!$response) {
+            return response()->json(['These credentials do not match our records.'], Response::HTTP_FORBIDDEN);
+        }
+
+        return response()->json(['message' => 'User created successfully'], Response::HTTP_CREATED);
```

Comment:
I have moved all response messages to external file located in resource/lang/en/messages.php, you might do it as well for clarity. 
It will help adding localization support in future.
You can then call them using helper __('filename.somearraykeys'), e.g. ['message'=>__('messages.users.created')]